python/lsst/metaserv/metaREST_v0.py

```diff
@@ -24,134 +24,134 @@
-    return runDbQueryM(
-        "SELECT DISTINCT lsstLevel FROM Repo WHERE repoType = 'db'",
-        None,
-        "No types with at least one database found.")
+    query = "SELECT DISTINCT lsstLevel FROM Repo WHERE repoType = 'db'"
+    message = "No types with at least one database found."
+    return _resultOf(text(query), message=message)
 
 @metaREST.route('/db/<string:lsstLevel>', methods=['GET'])
 def getDbPerType(lsstLevel):
     '''Lists databases for a given type.'''
-    # todo: this will currently fail if Repo table does not exist
-    return runDbQueryM(
-        "SELECT dbName FROM Repo JOIN DbMeta on (repoId=dbMetaId) "
-        "WHERE lsstLevel = %s",
-        (lsstLevel,),
-        "No database found.")
+    query = "SELECT dbName FROM Repo JOIN DbMeta on (repoId=dbMetaId) WHERE lsstLevel = :lsstLevel"
+    message = "No database found."
+    return _resultsOf(text(query), paramMap={"lsstLevel":lsstLevel}, message=message)
 
 @metaREST.route('/db/<string:lsstLevel>/<string:dbName>', methods=['GET'])
 def getDbPerTypeDbName(lsstLevel, dbName):
     '''Retrieves information about one database.'''
     # We don't use lsstLevel here because db names are unique across all types.
-    return runDbQuery1(
-        "SELECT Repo.*, DbMeta.* "
-        "FROM Repo JOIN DbMeta on (repoId=dbMetaId) "
-        "WHERE dbName = %s",
-        (dbName,),
-        "Database '%s' not found." % dbName)
+    query = "SELECT Repo.*, DbMeta.* " \
+            + "FROM Repo JOIN DbMeta on (repoId=dbMetaId) WHERE dbName = :dbName"
+    message = "No Repos in Database '%s' found." % dbName
+    return _resultOf(text(query), paramMap={"dbName":dbName}, message=message)
 
 @metaREST.route('/db/<string:lsstLevel>/<string:dbName>/tables', methods=['GET'])
 def getDbPerTypeDbNameTables(lsstLevel, dbName):
     '''Lists table names in a given database.'''
-    return runDbQueryM(
-        "SELECT table_name FROM information_schema.tables "
-        "WHERE table_schema= %s ",
-        (dbName,),
-        "No tables found in database '%s'." % dbName)
+    query = "SELECT table_name FROM information_schema.tables WHERE table_schema=:dbName"
+    message = "No tables found in database '%s'." % dbName
+    return _resultsOf(text(query), paramMap={"dbName":dbName}, message=message)
 
 @metaREST.route('/db/<string:lsstLevel>/<string:dbName>/tables/'
                 '<string:tableName>', methods=['GET'])
 def getDbPerTypeDbNameTablesTableName(lsstLevel, dbName, tableName):
     '''Retrieves information about a table from a given database.'''
-    return runDbQuery1(
-        "SELECT DDT_Table.* FROM DDT_Table JOIN DbMeta USING (dbMetaId) "
-        "WHERE dbName=%s AND tableName=%s",
-        (dbName, tableName),
-        "Table '%s.%s'not found." % (dbName, tableName))
+    query = "SELECT DDT_Table.* FROM DDT_Table " \
+            + "JOIN DbMeta USING (dbMetaId) " \
+            + "WHERE dbName=:dbName AND tableName=:tableName"
```

Comment:
String literal can be concatenated without '+' (it's faster as it's done at compile time).